4 note choreo

kmshah2 opened the pull request:
Created the 4 note auto using choreo.

kmshah2 committed:
non completed choreo paths. 4 note auto

kmshah2 committed:
finished 4 note choreo path

gcschmit commented:
Before we can test this in simulation or on the robot (and we should test in situation first), you need to add the auto command in RobotContainer. You can model it off of the other auto commands. Please let me know if you need a hand.

gcschmit committed:
Update robot constants for Choreo

I guessed for the mass and rotational inertia. Everything else matches the actual robot.

gcschmit committed:
Regenerated trajectories

Incorporating updated robot characteristics.

gcschmit commented:
Why is the path for the first of the three notes two different path and the other two a single path?
In addition, the Back-to-Speaker-2 path cannot be generated. I think this is because there are 3 way points, where 2 of them are on top of each other.

gcschmit commented:
I did notice that there is only one PathPlanner auto for the 5-note. Each Choreo path needs its own PathPlanner auto. We then build a command that includes each of the PathPlanner autos with conditionals and shooting in between.

kmshah2 committed:
added choreo path into robot code

kmshah2 committed:
fixed compile error with choreo path

lwgolobitsh committed:
troubleshoot choreo path

kmshah2 committed:
Fixed all paths to have note targeting, added shoot commands in code, fixed shoot command in choreo to Shoot At 4 Meters.

gcschmit committed:
tweak 5-note Choreo auto

* add velocity constraint when grabbing the first note
* remove stop constraint at end of path when grabbing first note since we don’t want to stop; we want to continue with the teleop swerve collect note command
* add command name (Shoot At 4 Meters) to event marker when grabbing first note and position at start of path since the command will wait until the robot is 4 meters from the speaker to shoot
* set shooting position to SOURCE_SIDE_AUTO_1 since that is the proper velocity and angle when shooting on the move at a distance of 4 meters
* invoke setShootingPosition after each shot to ensure shooter angle and wheels remain at proper settings throughout auto
* added 2.0 m/s velocity constraint for fourth_note_center
* add 2 second timeout when collecting notes near the speaker
* concerned that choreo paths reset the robot’s odometry at the start of each path but the robot may not be where intended due to the teleop swerve collect note command

gcschmit committed:
remove event marker for shooting when returning to speaker

gcschmit commented:
Pull in (indirectly) via #244
